compiler/qsc_parse/src/item.rs

```diff
@@ -98,10 +98,22 @@ pub(super) fn parse_fragments(s: &mut Scanner) -> Result<Vec<Fragment>> {
 }
 
 fn parse_fragment(s: &mut Scanner) -> Result<Fragment> {
-    if let Some(namespace) = opt(s, parse_namespace)? {
+    let doc = parse_doc(s);
+    if let Some(mut namespace) = opt(s, parse_namespace)? {
+        namespace.doc = doc.into();
         Ok(Fragment::Namespace(namespace))
     } else {
-        stmt::parse(s).map(Fragment::Stmt)
+        let kind = s.peek().kind;
+        let span = s.peek().span;
+        let mut fragment = stmt::parse(s).map(Fragment::Stmt)?;
+        if let Fragment::Stmt(stmt) = &mut fragment {
```

Comment:
This is always true.

Reply:
Good point. I thought about using a let else pattern but then I realized I could just skip the mapping on the line above. I'll fix that.